hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieMetadataWriteUtils.java

```diff
@@ -91,8 +108,9 @@ public static HoodieWriteConfig createMetadataWriteConfig(
         .withCleanConfig(HoodieCleanConfig.newBuilder()
             .withAsyncClean(DEFAULT_METADATA_ASYNC_CLEAN)
             .withAutoClean(false)
-            .withCleanerParallelism(parallelism)
-            .withCleanerPolicy(HoodieCleaningPolicy.KEEP_LATEST_COMMITS)
+            .withCleanerParallelism(defaultParallelism)
+            .withCleanerPolicy(HoodieCleaningPolicy.KEEP_LATEST_FILE_VERSIONS)
+            .retainFileVersions(2)
```

Comment:
This is a big behavior change, why we always leave some log files there then?

Reply:
I understand it could be a larger change, but file versions makes sense in general. If uber has been running w/ file versions for 6+ months, we should do a round of testing on our end, and can possibly proceed.
 but incremental cleaning may not kick in. so, for large MDTs, wondering will there be any latency hit

Reply:
Even if Uber has been running for 6+ months, it does not mean the config work well for OSS, because while we migrating the Uber patches, many fixes and other nuances are introduced, I would suggest we move this change to the next release to keep the stability of existing MDT workflow.

Reply:
I have reverted this change.